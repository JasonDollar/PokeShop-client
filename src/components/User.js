import { useQuery } from '@apollo/react-hooks'
import gql from 'graphql-tag'
import PropTypes from 'prop-types'


export const CURRENT_USER_QUERY = gql`
  query CURRENT_USER_QUERY {
    me {
      id
      name
      email 
      role
      offers {
        id
        name
        price
        pokemon {
          id
          image
        }
      }
      wallet {
        balance
      }
    }
  }
`

const User = props => {
  const payload = useQuery(CURRENT_USER_QUERY, {
    fetchPolicy: 'cache-and-network',
  })

  return props.children(payload)
}


export default User

User.propTypes = {
  children: PropTypes.any.isRequired,
}