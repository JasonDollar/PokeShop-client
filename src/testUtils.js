const fakeWallet = () => ({
  __typename: 'Wallet',
  id: '123213',
  balance: 100000,
  owner: '4234',
})

const fakeWallet2 = () => ({
  __typename: 'Wallet',
  id: 'dfghdfghdfhg',
  balance: 100000,
  owner: '4255',
})

const fakeUser = () => ({
  __typename: 'User',
  id: '4234',
  name: 'Miss Hilda Fahey',
  email: 'Satterfield.Horacio@gmail.com',
  role: 'admin',
  offers: [],
  cart: [],
  wallet: fakeWallet(),
})

const fakeUser2 = () => ({
  __typename: 'User',
  id: '4255',
  name: 'Mrs. Xzavier Mueller',
  email: 'Miller_Reichert@yahoo.com',
  role: 'admin',
  offers: [],
  cart: [],
  wallet: fakeWallet(),
})

const fakePokemon = () => ({
  __typename: 'Pokemon',
  id: 1,
  name: 'bulbasaur',
  url: 'https://pokeapi.co/api/v2/pokemon/1/',
  image: 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.pg',
  pokeType: ['grass'],
})

const fakePokemonOffer = () => ({
  __typename: 'PokemonOffer',
  id: '618726387213',
  name: 'bulbasaur',
  price: 1000,
  description: 'It is a fake pokemonOffer description',
  pokemon: fakePokemon(),
  seller: fakeUser(),
  createdAt: '2019-06-20T20:38:07.889+00:00',
})

const fakeCartItem = () => ({
  __typename: 'CartItem',
  id: '5d0d361471e9af00178faea2',
  quantity: 1,
  pokemon: fakePokemon(),
  user: fakeUser(),
})

const fakeOrderItem = (changes) => ({
  __typename: 'OrderItem',
  id: '5d0bef4b49cbfc0017dce192',
  quantity: 1,
  price: 1000,
  pokemon: fakePokemon(),
  user: fakeUser(),
  seller: fakeUser2(),
  ...changes,
})

const fakeOrder = () => ({
  __typename: 'Order',
  id: '5d0bef4b49cbfc0017dce193',
  price: 10000,
  user: fakeUser(),
  items: [fakeOrderItem(), fakeOrderItem({ id: '5d0bf3a549cbfc0017dce199' })],
})

export {
  fakeUser,
  fakeUser2,
  fakeWallet,
  fakePokemon,
  fakePokemonOffer,
  fakeCartItem,
  fakeOrderItem,
  fakeOrder,
}