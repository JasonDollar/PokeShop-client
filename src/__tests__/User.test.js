import React from 'react'
import { mount } from 'enzyme'
import { act } from 'react-dom/test-utils'
import toJSON from 'enzyme-to-json'
import wait from 'waait'
import { MockedProvider } from '@apollo/react-testing'
import User, { CURRENT_USER_QUERY } from '../components/User'
import { fakeUser } from '../testUtils'

const signedInMocks = [
  {
    request: { query: CURRENT_USER_QUERY },
    result: { data: { me: fakeUser() } },
  },
]

describe('<User/>', () => {
  test('should render child component', async () => {
    const Hey = () => <p>Heyyy</p>
    const wrapper = mount(
      <MockedProvider mocks={signedInMocks}>
        <User>
          {data => <Hey />}
        </User>
      </MockedProvider>,
    )
    await act(async () => {
      await wait(0)
      wrapper.update()
    })
    
    expect(toJSON(wrapper.find('Hey'))).toMatchSnapshot()
  })
})