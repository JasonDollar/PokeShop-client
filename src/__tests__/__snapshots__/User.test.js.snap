// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<User/> should render child component 1`] = `
<Hey>
  <p>
    Heyyy
  </p>
</Hey>
`;
